File: examples/convert.py
"""
FileConvertLab API - Python Example
https://fileconvertlab.com/developers/api
"""

import requests
import time
import sys

API_KEY = 'sk_live_your_api_key_here'
BASE_URL = 'https://fileconvertlab.com/api/v1'


def convert_file(input_path: str, operation: str, output_path: str):
    """
    Convert a file using FileConvertLab API.

    Args:
        input_path: Path to the input file
        operation: Conversion operation (e.g., 'pdf-to-word', 'word-to-pdf')
        output_path: Path to save the converted file
    """
    headers = {'X-API-Key': API_KEY}

    # Create conversion job
    print(f'Uploading {input_path}...')
    with open(input_path, 'rb') as f:
        response = requests.post(
            f'{BASE_URL}/jobs',
            headers=headers,
            files={'file': f},
            data={'operation': operation}
        )

    if response.status_code != 202:
        print(f'Error: {response.json()}')
        return False

    job = response.json()
    print(f'Job created: {job["id"]}')

    # Poll for completion
    print('Processing', end='', flush=True)
    while job['status'] in ('pending', 'processing'):
        time.sleep(1)
        print('.', end='', flush=True)
        response = requests.get(
            f'{BASE_URL}/jobs/{job["id"]}',
            headers=headers
        )
        job = response.json()

    print()

    if job['status'] == 'failed':
        print(f'Conversion failed: {job.get("error", "Unknown error")}')
        return False

    # Download result
    print(f'Downloading to {output_path}...')
    response = requests.get(
        f'{BASE_URL}{job["result"]["download_url"]}',
        headers=headers
    )

    with open(output_path, 'wb') as f:
        f.write(response.content)

    print(f'Done! Saved to {output_path}')
    return True


if __name__ == '__main__':
    if len(sys.argv) != 4:
        print('Usage: python convert.py <input_file> <operation> <output_file>')
        print('Example: python convert.py document.pdf pdf-to-word document.docx')
        sys.exit(1)

    convert_file(sys.argv[1], sys.argv[2], sys.argv[3])

File: examples/convert.js
/**
 * FileConvertLab API - Node.js Example
 * https://fileconvertlab.com/developers/api
 */

const fs = require('fs');
const path = require('path');

const API_KEY = 'sk_live_your_api_key_here';
const BASE_URL = 'https://fileconvertlab.com/api/v1';

async function convertFile(inputPath, operation, outputPath) {
  const headers = { 'X-API-Key': API_KEY };

  // Create conversion job
  console.log(`Uploading ${inputPath}...`);

  const formData = new FormData();
  const fileBuffer = fs.readFileSync(inputPath);
  const blob = new Blob([fileBuffer]);
  formData.append('file', blob, path.basename(inputPath));
  formData.append('operation', operation);

  let response = await fetch(`${BASE_URL}/jobs`, {
    method: 'POST',
    headers,
    body: formData,
  });

  if (response.status !== 202) {
    const error = await response.json();
    console.error('Error:', error);
    return false;
  }

  let job = await response.json();
  console.log(`Job created: ${job.id}`);

  // Poll for completion
  process.stdout.write('Processing');
  while (job.status === 'pending' || job.status === 'processing') {
    await new Promise((r) => setTimeout(r, 1000));
    process.stdout.write('.');

    response = await fetch(`${BASE_URL}/jobs/${job.id}`, { headers });
    job = await response.json();
  }
  console.log();

  if (job.status === 'failed') {
    console.error(`Conversion failed: ${job.error || 'Unknown error'}`);
    return false;
  }

  // Download result
  console.log(`Downloading to ${outputPath}...`);
  response = await fetch(`${BASE_URL}${job.result.download_url}`, { headers });
  const buffer = Buffer.from(await response.arrayBuffer());
  fs.writeFileSync(outputPath, buffer);

  console.log(`Done! Saved to ${outputPath}`);
  return true;
}

// CLI usage
const args = process.argv.slice(2);
if (args.length !== 3) {
  console.log('Usage: node convert.js <input_file> <operation> <output_file>');
  console.log('Example: node convert.js document.pdf pdf-to-word document.docx');
  process.exit(1);
}

convertFile(args[0], args[1], args[2]);
